Source/carthage/Build.swift

```diff
@@ -58,6 +59,7 @@ public struct BuildCommand: CommandProtocol {
 				<*> ColorOptions.evaluate(m)
 				<*> m <| Option(key: "verbose", defaultValue: false, usage: "print xcodebuild output inline")
 				<*> m <| Option(key: "project-directory", defaultValue: FileManager.default.currentDirectoryPath, usage: "the directory containing the Carthage project")
+				<*> m <| Option(key: "log-path", defaultValue: BuildCommand.defaultLogPath, usage: "path to the xcode build output. A temporary file is used by default")
```

Comment:
I initially had that approach but then used this one for the PR because: 

1. It makes passing the value into Build from Update operations easier without duplicating the default string value
2. avoiding `mktemps` or documenting that `mktemps` will happen on input. When a custom log is used. A user would be surprised if they passed `custom-log-for-project-X` and the logs appeared in `custom-log-for-project-6`

I can move it back to optional to cleanup the help text, but I still think the equality check is handy. If it goes away we need to make the help text longer and explain the special behavior around `X` in paths.